lib/matplotlib/projections/polar.py

```diff
@@ -240,6 +242,146 @@ def zoom(self, direction):
         return self.base.zoom(direction)
 
 
+class ThetaTick(maxis.XTick):
+    """
+    A theta-axis tick.
+
+    This subclass of `XTick` provides angular ticks with some small
+    modification to their re-positioning such that ticks are rotated based on
+    tick location. This results in ticks that are correctly perpendicular to
+    the arc spine.
+
+    Labels are also rotated to be parallel to the spine. The label padding is
+    also applied here since it's not possible to use a generic axes transform
+    to produce tick-specific padding.
+    """
+    def __init__(self, axes, *args, **kwargs):
+        self._text1_translate = mtransforms.ScaledTranslation(
+            0, 0,
+            axes.figure.dpi_scale_trans)
+        self._text2_translate = mtransforms.ScaledTranslation(
+            0, 0,
+            axes.figure.dpi_scale_trans)
+        maxis.XTick.__init__(self, axes, *args, **kwargs)
+
+    def _get_text1(self):
+        t = maxis.XTick._get_text1(self)
+        t.set_rotation_mode('anchor')
+        t.set_transform(t.get_transform() + self._text1_translate)
+        return t
+
+    def _get_text2(self):
+        t = maxis.XTick._get_text2(self)
+        t.set_rotation_mode('anchor')
+        t.set_transform(t.get_transform() + self._text2_translate)
+        return t
+
+    def _apply_params(self, **kw):
+        maxis.XTick._apply_params(self, **kw)
```

Comment:
Why not `super(ThetaTick, self)._apply_params(**kw)`?

Reply:
No reason; just couldn't remember which version was okay with old Python.